encoding/codecv2/codecv2.go

```diff
@@ -183,6 +184,14 @@ func ConstructBlobPayload(chunks []*encoding.Chunk, useMockTxData bool) (*kzg484
 		return nil, common.Hash{}, nil, err
 	}
 
+	// Only apply this check when the uncompressed batch data has exceeded 128KiB.
+	if len(blobBytes) > 131072 {
+		// check compressed data compatibility
+		if err = encoding.CheckCompressedDataCompatibility(compressedBlobBytes); err != nil {
+			return nil, common.Hash{}, nil, err
```

Comment:
I think we can embed it here since it's encoding's check instead of batch-proposer.
btw, is there anything suspicious about batch-proposer? maybe I have ignored it.

Reply:
discussed with @noel2004, will apply this fix: https://github.com/scroll-tech/da-codec/pull/16#discussion_r1667594324

Reply:
moved the check to `EstimateChunkL1CommitBatchSizeAndBlobSize` and `EstimateBatchL1CommitBatchSizeAndBlobSize` in this commit: [a554fe4](https://github.com/scroll-tech/da-codec/pull/16/commits/a554fe417a393500de4f7545c6a72bb10a3351ff), pls review. also cc @noel2004

Reply:
OK, looks good to me. If you're confident that adding this check here ensures that all chunks and batches pass the `CheckCompressedDataCompatibility` check, then this should be good.

Reply:
So again i emphased that we only need this check when uncompressed bytes exceed 128kB